File: doric-flutter/doric/lib/doric/plugin/network_plugin.dart
import 'dart:convert';

import 'package:dio/dio.dart';
import 'package:doric/doric/doric_context.dart';
import 'package:doric/doric/doric_promise.dart';
import 'package:doric/doric/engine/doric_plugin.dart';
import 'package:doric/flutter_jscore.dart';

class NetworkPlugin extends DoricPlugin {
  NetworkPlugin(DoricContext context) : super(context);
  var dio = new Dio();

  @override
  Map initFunc() {
    return {
      "request": request,
    };
  }

  @override
  void onTearDown() {}

  void request(JSValue jsValue, DoricPromise promise) {
    var data = jsValue.toObject().toMap();
    var method = data["method"];
    var header = data["header"] ?? {};
    header["Content-Type"] =
        header["Content-Type"] ?? "application/json; charset=utf-8";

    dio
        .request(data["url"],
            options: Options(method: method, headers: data["header"]))
        .then((value) {
          var result={
            "status":value.statusCode,
            "headers":json.encode(value.headers.map),
            "data":value.data.toString()
          };
      promise.resolve([ JSValue.makeFromObject(getDoricContext().getJSContext(), result)]);
    }).catchError((e) {
      promise.reject([JSValue.makeString(getDoricContext().getJSContext(), e.toString())]);
    });
  }

  void get() {}

  void post() {}
}

File: doric-ohos/doric-ohos/src/main/ets/components/plugin/NetworkPlugin.ts
import http from '@ohos.net.http'

import { DoricPlugin } from '../lib/sandbox'

export class NetworkPlugin extends DoricPlugin {
  request(props: any) {
    let method = "GET"
    switch (props.method) {
      case "get":
        method = "GET"
        break
      case "post":
        method = "POST"
        break
      case "put":
        method = "PUT"
        break
      case "delete":
        method = "DELETE"
        break
    }

    return new Promise((resolve, reject) => {
      http.createHttp().request(
        props.url,
        {
          method: method as any,
          header: props.headers ?? undefined,
          extraData: props.data ?? undefined,
          connectTimeout: props.timeout ?? undefined
        },
        (error, data) => {
          if (error) {
            reject(error)
          } else {
            resolve(data)
          }
        }
      )
    })
  }
}